File: app/src/main/java/im/kirillt/yandexmoneyclient/events/download/DownloadAllEvent.java
package im.kirillt.yandexmoneyclient.events.download;

import android.content.Context;


public class DownloadAllEvent implements DownloadEvent{
    private DownloadAccountInfoEvent downloadAccountInfoEvent;
    private DownloadHistoryEvent downloadHistoryEvent;

    public DownloadAllEvent(Context context) {
        downloadAccountInfoEvent = new DownloadAccountInfoEvent(context);
        downloadHistoryEvent = new DownloadHistoryEvent(context);
    }

    public void download() {
        downloadAccountInfoEvent.download();
        downloadHistoryEvent.download();
    }
}

File: app/src/main/java/im/kirillt/yandexmoneyclient/events/download/DownloadAllEvent.kt
package im.kirillt.yandexmoneyclient.events.download

import android.content.Context
import im.kirillt.yandexmoneyclient.YMCApplication


class DownloadAllEvent(context: Context) : DownloadEvent {
    private val downloadAccountInfoEvent: DownloadAccountInfoEvent
    private val downloadHistoryEvent: DownloadHistoryEvent

    init {
        downloadAccountInfoEvent = DownloadAccountInfoEvent(context, YMCApplication.getLogin())
        downloadHistoryEvent = DownloadHistoryEvent(context)
    }

    override fun download() {
        downloadAccountInfoEvent.download()
        downloadHistoryEvent.download()
    }
}
